lib/python/qmk/questions.py

```diff
@@ -57,17 +57,23 @@ def yesno(prompt, *args, default=None, **kwargs):
             return False
 
 
-def question(prompt, *args, default=None, confirm=False, answer_type=str, **kwargs):
+def question(prompt, *args, default=None, confirm=False, answer_type=str, validate=None, **kwargs):
     """Prompt the user to answer a question with a free-form input.
 
+    Arguments:
         prompt
-            The prompt to present to the user. Can include ANSI and format strings like milc's `cli.print()`.
+            The prompt to present to the user. Can include ANSI and format strings like milc's `cli.echo()`.
 
         default
             The value to return when the user doesn't enter any value. Use None to prompt until they enter a value.
 
         answer_type
             Specify a type function for the answer. Will re-prompt the user if the function raises any errors. Common choices here include int, float, and decimal.Decimal.
+
+        validate
+            This is an optional function that can be used to validate the answer. It should return True or False and have the following signature:
+
+                def function_name(answer, *args, **kwargs):
```

Comment:
Maybe we could add something like this:
```
For simple scenarios a lambda function works, too:

    lambda answer: True if answer <condition> else False
```

Reply:
I'd prefer not promoting the use of lambdas. I find that often those sorts of simple checks are useful in multiple places.